src/XProfile/VirtoCommerce.XProfile/Schemas/ProfileSchema.cs

```diff
@@ -366,51 +394,54 @@ firstName lastName
                         }
                          */
 #pragma warning restore S125 // Sections of code should not be commented out
+
+            #endregion
             _ = schema.Mutation.AddField(FieldBuilder.Create<object, IdentityResult>(typeof(IdentityResultType))
                      .Name("updateRole")
                      .Argument<NonNullGraphType<InputUpdateRoleType>>(_commandName)
                      .ResolveAsync(async context =>
                      {
                          var command = context.GetArgument<UpdateRoleCommand>(_commandName);
-                         await CheckAuthAsync(command.UserId, command, PlatformConstants.Security.Permissions.SecurityUpdate);
+                         await CheckAuthAsync(context.GetCurrentPrincipal(), command, PlatformConstants.Security.Permissions.SecurityUpdate);
 
                          return await _mediator.Send(command);
                      })
                      .FieldType);
         }
 
 
-
-        private async Task CheckAuthAsync(string userId, object resource, params string[] permissions)
+        private async Task CheckAuthAsync(ClaimsPrincipal userPrincipal, object resource, params string[] permissions)

```

Comment:
Seems that `ClaimsPrincipal` parameter is redundant, as imho in all places `context.GetCurrentUser()` is used. Correct me if I'm wrong. 
So I think we could remove this param.